From 8f8fac5d958520b192fae5b08ac61a19e450841c Mon Sep 17 00:00:00 2001
From: Gregory Lee <grlee77@gmail.com>
Date: Wed, 8 Dec 2021 13:22:30 -0500
Subject: [PATCH] relax test condition to make it more robust to variable CI
 load

This should resolve occasional sporadic test failures seen recently.
---
 skimage/_shared/tests/test_coord.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/skimage/_shared/tests/test_coord.py b/skimage/_shared/tests/test_coord.py
index cd1749fa27..58b8fc7af1 100644
--- a/skimage/_shared/tests/test_coord.py
+++ b/skimage/_shared/tests/test_coord.py
@@ -75,7 +75,11 @@ def test_max_batch_size():
                    max_split_size=20000)
     dur2 = time.time() - tstart
 
-    assert dur1 < dur2
+    # Originally checked dur1 < dur2 to assert that the default batch size was
+    # faster than a much larger batch size. However, on rare occasion a CI test
+    # case would fail with dur1 ~5% larger than dur2. To be more robust to
+    # variable load or differences across architectures, we relax this here.
+    assert dur1 < 1.33 * dur2
 
 
 @pytest.mark.parametrize("p", [1, 2, np.inf])
